src/task2/damage_selection_gui/damageselector.py

```diff
@@ -0,0 +1,327 @@
+        self.load_next_img_or_end()
+
+    def left_mouse_down(self, event):
+        """Called whenever the left mouse is clicked on the image canvas"""
+        x, y = self.movable_image.win_to_cv((event.x, event.y))
+        pass
+
+    def left_mouse_move(self, event):
+        """Called whenever left mouse is moved while being held down
+        on canvas.
+        """
+        x, y = self.movable_image.win_to_canvas((event.x, event.y))
+        self._clear_canvas()
+        if self.__state == State.CIRCLE:
+            self._draw_circle_on_canvas_from_last_click((x, y))
+        elif self.__state == State.POLLY:
+            self._draw_line_on_canvas_from_last_click((x, y))
+
+    def _clear_canvas(self):
+        self.movable_image.canvas.delete(MARK_TAG)
+
+    # The following to methods do not directly edit the final image.
+    # They add lines and circles to the canvas to preview the next part of the
+    # selection.
+
+    def _draw_line_on_canvas_from_last_click(self, point: Tuple[int, int]):
+        """Draws a line from the last point in the clicks list to a point on
+        the point given on the canvas. <point> is in canvas cords.
+        """
+        if len(self.__clicks) >= 1:
+            p1 = self.movable_image.cv_to_canvas(self.__clicks[-1])
+            p2 = point
+            self.movable_image.canvas.create_line(p1[0], p1[1],
+                                                  p2[0], p2[1],
+                                                  width=CANVAS_LINE_WIDTH,
+                                                  tags=MARK_TAG,
+                                                  dash=CANVAS_DASH_PATTERN,
+                                                  fill=CANVAS_LINE_COLOR)
+
+    def _draw_circle_on_canvas_from_last_click(self, point: Tuple[int, int]):
+        """Draws a circle with its edge being the last point in the clicks list
+        and its center being the point provided on the canvas.
+        <point> is in canvas cords.
+        """
+        if len(self.__clicks) >= 1:
+            p1 = self.movable_image.cv_to_canvas(self.__clicks[-1])
+            p2 = point
+            r = self._dist(p1, p2)
+            self.movable_image.canvas.create_oval(p2[0] - r, p2[1] - r,
+                                                  p2[0] + r, p2[1] + r,
+                                                  width=CANVAS_LINE_WIDTH,
+                                                  tags=MARK_TAG,
+                                                  dash=CANVAS_DASH_PATTERN,
+                                                  outline=CANVAS_LINE_COLOR)
+
+    def left_mouse_up(self, event):
+        """Called when ever left mouse button is released"""
+        x, y = self.movable_image.win_to_cv((event.x, event.y))
+
+        if self.__state == State.POLLY:
+            self._draw_line_on_cv_from_last_click((x, y))
+            self.__clicks.append((x, y))
+
+            if len(self.__clicks) >= 4:
```

Comment:
Not a requested change but more of a comment - nested if statements that are huggeee are ugly af. We can reverse this logic to 

```python
if len(self.__clicks) < 4:
    return

self._draw_line_on_cv_from_last_click(self.__clicks[0])
...
```

to short-circuit quicker to prevent large blocks of code in if statements. A nice way to `kick out the invalid conditions before they bite you in the ass later on`. Multiple exit points imo are much better than a single exit point but at a sacrifieced readability of code https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement